overlord/snapstate/handlers.go

```diff
@@ -836,16 +838,21 @@ func (m *SnapManager) doUnlinkCurrentSnap(t *state.Task, _ *tomb.Tomb) error {
 
 	snapst.Active = false
 
-	// do the final unlink
-	linkCtx := backend.LinkContext{
-		FirstInstall: false,
-		// This task is only used for unlinking a snap during refreshes so we
-		// can safely hard-code this condition here.
-		RunInhibitHint: runinhibit.HintInhibitedForRefresh,
-	}
-	err = m.backend.UnlinkSnap(oldInfo, linkCtx, NewTaskProgressAdapterLocked(t))
-	if err != nil {
-		return err
+	if oldInfo.Type() != snap.TypeSnapd {
```

Comment:
do we test the skipping of this, I don't see a new test that mentions unlink-current-snap

Reply:
Added now.